src/NuGet.Clients/NuGet.PackageManagement.UI/Utility/PackageSourceMappingUtility.cs

```diff
@@ -91,5 +91,52 @@ internal static class PackageSourceMappingUtility
 
             mappingProvider.SavePackageSourceMappings(newAndExistingPackageSourceMappingItems);
         }
+
+        internal static string? GetNewSourceMappingSourceName(PackageSourceMapping packageSourceMapping, PackageSourceMoniker activePackageSourceMoniker)
+        {
+            string? sourceMappingSourceName = packageSourceMapping.IsEnabled
+                && !activePackageSourceMoniker.IsAggregateSource
+                ? activePackageSourceMoniker.PackageSourceNames.First() : null;
+
+            return sourceMappingSourceName;
+        }
+
+        internal static void GetNewSourceMappingsFromAddedPackages(ref Dictionary<string, SortedSet<string>>? newSourceMappings, string newMappingSourceName, List<AccessiblePackageIdentity> added, PackageSourceMapping packageSourceMapping)
+        {
+            if (newMappingSourceName is null || added.Count == 0 || packageSourceMapping is null)
+            {
+                return;
+            }
+
+            List<string> addedPackagesWithNoSourceMappings = added.Select(_ => _.Id)
```

Comment:
I have never seen this pattern to use a throw away in Linq when getting the property.  This looks more _normal_ to me.  Just curious what everyone else's thoughts on this pattern?

```suggestion
            List<string> addedPackagesWithNoSourceMappings = added.Select(i => i.Id)
```

Reply:
Good callout! I've been trying to use throw-aways more often, when I don't care about the variable name and don't really need one. To me, arbitrary letters are unnecessary noise. (`i` probably isn't what I'd choose, but maybe `p` for "package" or the word itself?)

Reply:
It did look a bit unusual at first, but ultimately I find the code readable.

Reply:
From the docs of the C# language feature "discards": https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/discards

> Discards are placeholder variables that are intentionally unused in application code.

also

> Discards make the intent of your code clear. A discard indicates that our code never uses the variable.

Therefore, doing `_.Id` violates that intent.

The reason it works is because 1. when defining lamba functions, it's not necessary to specify types, when the compiler is capable of inferring the type (which it is in this code snippet). 2. C# allowed variables named `_` long before the discard feature was created, and to minimize risk of breaking customers with existing codebases, the language/compiler team continue to allow variables named `_`.

Therefore, this is not actually using the discard language feature, it's creating a variable named `_`.

Also from the docs:

> When _ is a valid discard, attempting to retrieve its value or use it in an assignment operation generates compiler error CS0103, "The name '_' doesn't exist in the current context". This error is because _ isn't assigned a value, and may not even be assigned a storage location.

So, depending on how familiar someone is with the discard language feature, using `_` as a variable name could be confusing (why didn't the compiler error out with CS0103?). Following the usual "use meaningful names" guideline (which I see isn't actually in our coding guidelines document) suggests that `_` isn't a good variable name as I can't infer any information at all about the data type or meaning of the variable from the name alone.

```suggestion
            List<string> addedPackagesWithNoSourceMappings = added.Select(package => package.Id)
```

though I'd expect most people would accept `p` as an acceptable variable name too. I think Jeff's suggestion `i` is not a good suggestion, because it further obfuscates what `i` is (is it an integer/index? is it a source? is it a package?).

Reply:
Thanks for explaining that. I realize now I started this habit to avoid redundant naming, and it's actually just been an underscore variable name all along 😁.